### PR TITLE
chore: generated tsdocs

### PR DESCRIPTION
**Pull request by shahednasser**
Generated TSDocs for the past release.

Note: I haven't updated examples as the examples are for modules without a public reference yet, so the examples can wait.

### CODE REVIEW

File: packages/types/src/auth/common/auth-user.ts
Comment (shahednasser) [inline packages/types/src/auth/common/auth-user.ts:1]: @pKorsholm can I get you to look at the TSDocs in this and all auth/authentication files? I'm not very confident in some of them. I also left some of them as generated by the docblock tool `{summary}` as i wasn't sure what the replacement should be. Feel free to directly edit or leave a comment for changes.

File: packages/types/src/customer/mutations.ts [hunk 1]
```diff
@@ -1,101 +1,402 @@
...
   created_by?: string
 }
 
+/**
+ * @privateRemark
+ *
+ * This seems to be the same as {@link CustomerGroupUpdatableFields} but with
+ * a typo?
+ */
 export interface CustomerGroupUpdatableFileds {
```
Comment (shahednasser) [inline packages/types/src/customer/mutations.ts]: @srindom can you take a look at the comment here?

Comment (srindom) [reply packages/types/src/customer/mutations.ts]: This probably snuck in here during a merge - you can delete this entirely

File: packages/types/src/auth/common/provider.ts [hunk 2]
```diff
@@ -1,22 +1,88 @@
...
   authUser?: any
+
+  /**
+   * The error message, if an error occurs.
+   */
   error?: string
+
+  /**
+   * {summary}
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Redirect location. Location takes precedence over success.
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 3]
```diff
@@ -1,22 +1,88 @@
...
   query: Record<string, string>
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
   body: Record<string, string>
+
+  /**
+   * {summary}
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Scope for authentication request
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 4]
```diff
@@ -1,22 +1,88 @@
...
   authScope: string
+
+  /**
+   * {summary}
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Protocol of incoming authentication request (ex. https)
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 5]
```diff
@@ -1,22 +1,88 @@
...
   headers: Record<string, string>
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
   query: Record<string, string>
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Body of incoming authentication request
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 6]
```diff
@@ -1,22 +1,88 @@
...
   url: string
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
   headers: Record<string, string>
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   *  Query params of incoming authentication request
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 7]
```diff
@@ -1,22 +1,88 @@
...
+ */
 export type AuthenticationInput = {
+  /**
+   * {summary}
+   */
   url: string
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Headers of incoming authentication request
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 8]
```diff
@@ -1,22 +1,88 @@
...
 
+/**
+ * @interface
+ *
+ * {summary}
+ */
 export type AuthenticationInput = {
+  /**
+   * {summary}
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * url of incoming authentication request
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 9]
```diff
@@ -1,22 +1,88 @@
...
+ * @interface
+ *
+ * {summary}
+ */
 export type AuthProviderScope = Record<string, unknown>
 
+/**
+ * @interface
+ *
+ * {summary}
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
 * Input for authentication and callback validation methods
```

File: packages/types/src/auth/common/provider.ts [hunk 10]
```diff
@@ -1,22 +1,88 @@
...
+ */
 export type AuthModuleProviderConfig = {
+  /**
+   * {summary}
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Provider name
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 11]
```diff
@@ -1,22 +1,88 @@
...
   name: string
+
+  /**
+   * Construct a type with a set of properties K of type T
+   */
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
  /**
   * Scope configurations for the provider
   */
```

File: packages/types/src/auth/common/provider.ts [hunk 12]
```diff
@@ -1,22 +1,88 @@
...
   scopes: Record<string, AuthProviderScope>
 }
 
+/**
+ * @interface
+ *
+ * {summary}
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
 *
 * Configuration of a single provider scope
```

File: packages/types/src/auth/common/provider.ts [hunk 13]
```diff
@@ -1,22 +1,88 @@
...
+  /**
+   * {summary}
+   */
   location?: string
 }
 
+/**
+ * @interface
+ *
+ * {summary}
```
Comment (pKorsholm) [inline packages/types/src/auth/common/provider.ts]:
```suggestion
 * Provider configuration for the Medusa auth module
```

File: packages/types/src/authentication/common/provider.ts [hunk 14]
```diff
@@ -1,22 +1,93 @@
+/**
```
Comment (pKorsholm) [inline packages/types/src/authentication/common/provider.ts]: This file is a merge error and should just be deleted 😄

File: packages/types/src/auth/common/auth-user.ts [hunk 15]
```diff
@@ -1,34 +1,128 @@
...
+   */
   id: string
+
+  /**
+   * The provider of the auth user.
+   */
   provider: string
+
+  /**
+   * The associated entity's ID.
```
Comment (pKorsholm) [inline packages/types/src/auth/common/auth-user.ts]:
```suggestion
   * The associated providers entity's ID.
```

**Review by pKorsholm — COMMENTED** [review]
I've left comments on the files in `/auth` with suggestions, the `/authentication` is a merge error from when we renamed the auth module and should just be removed 😄

File: packages/types/src/auth/service.ts [hunk 16]
```diff
@@ -11,52 +11,211 @@ import { Context } from "../shared-context"
...
 export type JWTGenerationOptions = {
   expiresIn?: string | number
 }
-
+/**
+ * The main service interface for the authentication module.
+ */
 export interface IAuthModuleService extends IModuleService {
+  /**
+   * This method represents the completion of an asynchronous operation
```
Comment (pKorsholm) [inline packages/types/src/auth/service.ts]:
```suggestion
   * Handle the incoming authentication request. This method is the first invoked method of the authentication flow.
```

File: packages/types/src/auth/service.ts [hunk 17]
```diff
@@ -11,52 +11,211 @@ import { Context } from "../shared-context"
...
+   * @example
+   * {example-code}
+   */
   authenticate(
     provider: string,
     providerData: AuthenticationInput
   ): Promise<AuthenticationResponse>
 
+  /**
+   * This method Represents the completion of an asynchronous operation
```
Comment (pKorsholm) [inline packages/types/src/auth/service.ts]:
```suggestion
   * Handle callback from an authentication provider. When an authentication has been initialized and the user is redirected this method handles the callback.
```

File: packages/types/src/auth/service.ts [hunk 18]
```diff
@@ -11,52 +11,211 @@ import { Context } from "../shared-context"
...
+   *
+   * @param {string} provider - {summary}
+   * @param {AuthenticationInput} providerData - {summary}
+   * @returns {Promise<AuthenticationResponse>} Represents the completion of an asynchronous operation
```
Comment (pKorsholm) [inline packages/types/src/auth/service.ts]:
```suggestion
   * @param {string} provider - The provider to use for callback validation.
   * @param {AuthenticationInput} providerData - The authentication data necessary to pass to the provider
   * @returns {Promise<AuthenticationResponse>} The authentication's result.
```

**Review by pKorsholm — APPROVED** [review]
auth part looks good to me

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**